File: backend/Domains/FlashCardSet.cs
namespace LumiLearn.Domains
{
    public class FlashCardSet
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public Guid LessonId { get; set; }
        
        // Navigation Properties
        public Lesson Lesson { get; set; }

        public ICollection<FlashCard> FlashCards { get; set; }
    }
}

File: frontend/types/flashcardSet.ts
import { CreateFlashcardRequest } from "./flashcard";

export type FlashcardSetOverview = {
    id: string;
    lessonId: string;
    title: string;
};

export type CreateFlashcardSetRequest = {
    lessonId: string;
    title: string;
    flashcards: CreateFlashcardRequest[];
};